src/Agent.Worker/JobRunner.cs

```diff
@@ -126,10 +126,11 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                         {
                             notSupportNet6Message = $"The operating system the agent is running on is \"{systemId}\" ({systemVersion}), which has not been tested with the .NET 6 based v3 agent. The v2 agent wil not automatically upgrade to the v3 agent. You can manually download the .NET 6 based v3 agent from https://github.com/microsoft/azure-pipelines-agent/releases. See https://aka.ms/azdo-pipeline-agent-version";
                         }
-
+ 
                         if (!string.IsNullOrWhiteSpace(notSupportNet6Message))
                         {                            
-                            if(!AgentKnobs.AcknowledgeNoUpdates.GetValue(jobContext).AsBoolean())
+                            if(AgentKnobs.AgentFailOnIncompatibleOS.GetValue(jobContext).AsBoolean() && 
```

Comment:
[possibly for future PR] Do we have telemetry to detect what systems/customers that would receive the warn (e.g. this may affect other RH derived OS's)?  That would be helpful for monitoring the rollout

Reply:
@merlynomsft we are working on it as part of another PR